clang/test/CheckedC/parsing/valid-where-clause.c

```diff
@@ -34,7 +33,7 @@ void valid_cases_nullstmt(_Nt_array_ptr<char> p, _Nt_array_ptr<char> q,
 
 int f();
 // Test where clauses on variable declarations inside a function.
-void valid_cases_decl(_Nt_array_ptr<char> p, _Nt_array_ptr<char> q) {
+void valid_cases_decl(_Nt_array_ptr<char> p, _Nt_array_ptr<char> q) { // expected-error {{it is not possible to prove that the inferred bounds of 'p' imply the declared bounds of 'p' after initialization}} expected-error {{it is not possible to prove that the inferred bounds of 'p' imply the declared bounds of 'p' after initialization}} expected-error {{it is not possible to prove that the inferred bounds of 'p' imply the declared bounds of 'p' after initialization}} expected-error {{it is not possible to prove that the inferred bounds of 'q' imply the declared bounds of 'q' after initialization}}
```

Comment:
Could we please discuss what exactly is happening here and analyze the compiler error messages?